# App profiles for Google Play and App Store
Our aim in this project is to find mobile apps that are profitable to App Store and Google Play. We build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app.

## Opening and exploring the Data

[A data set](https://www.kaggle.com/lava18/google-play-store-apps/home) Containing data about approximately Android Apps from Google Play

[A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) Containing data about ios Apps from App Store.

Lets Start by opening two data sets and then continue with exploring the data

In [1]:
##App Store
openAppleStore = open("Downloads/app-store-apple-data-set-10k-apps/AppleStore.csv", encoding="utf8")
from csv import reader
readAppStore = reader(openAppleStore)
appleDataSet = list(readAppStore)
iosHeader = appleDataSet[0]
ios = appleDataSet[1:]


## Google Play Store
openGooglePlay = open("Downloads/google-play-store-apps/googleplaystore.csv", encoding="utf8")
from csv import reader
readGooglePlay = reader(openGooglePlay)
googleDataSet = list(readGooglePlay)
androidHeader = googleDataSet[0]
android = googleDataSet[1:]

To make it easier to explore the two data sets, we will first write a function named explore_Data() that we can use repeatedly to explore rows in a more readable way, we will also introduce a parameter rows_and_columns, which can be used to show the number of rows and columns 

In [2]:
def explore_Data(data_set, start, end, rows_and_columns= False):
    dataSet = data_set[start:end]
    for row in dataSet:
        print(row)
        print("\n")
    if rows_and_columns:
        print("Number of rows ",len(data_set))  
        print("Number of columns ", len(data_set[0]))  

In [3]:
print(androidHeader)
print("\n")
explore_Data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows  10841
Number of columns  13


Now, we can see that we Play store has 10841 apps, and 13 Columns.

In [4]:
print(iosHeader)
print("\n")
explore_Data(ios, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows  7197
Number of columns  17


Now, we can see we have 7197 rows, and 17 columns.

# Deleting wrong Data

The Google Play Dataset has a [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), and we can see in one of the [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015), they outline a error for row 10472, let's just print that row and compare to other row.

In [5]:
print(android[10472]) # printing 10472 row
print(androidHeader)  # printing Android Header
print(android[1])  # printing 1st row for comparison

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


The row 10472 corresponds to 'Life Made WI-Fi Touchscreen Photo Frame', and we can clearly see that rating is '19', which is clearly wrong, as maximum rating for Google PlayStore is 5, So we need to delete this row. 

In [6]:
print(len(android)) 
del android[10472]  #deleting the wrong row
print(len(android))

10841
10840


If we go through Google Play data, we can see some apps are being repeated, For example Instagram has four entries.

In [7]:
for app in android:
    if "Instagram"== app[0]:
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [8]:
unique_apps = []
duplicate_apps = []
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name);
    else:
        unique_apps.append(name)
print("unique_apps", len(unique_apps));
print("/n");
print("duplicate_apps", len(duplicate_apps));
print(duplicate_apps[:15])

unique_apps 9659
/n
duplicate_apps 1181
['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


As, we can see there are 1181 cases where apps are being repeated.
We dont't want to count apps more than once, when we analyze data. So, we need to remove duplicate entries and keep only one entry per app. One way which we could do is deleting randomly, but we can find a better way.

If, we can examine the data which we have printed for Instagram, we can see rows are pretty much same, but the main difference happens in 4th column, i.e Reviews. The different number shows these data where collected at different times. We can use this to build a criterion for keeping rows. We won't remove rows randomly, but rather we'll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings. 

To do that, we will :


1) Create a dictionary, where each key is a unique app name, and value is the highest number of reviews of that app.


2) Use the dictionary to create a new data set, which will have only one entry per app (and we only select the apps with the highest number of reviews)

In [9]:
review_max = {}
for row in android:
    name  = row[0]
    n_reviews = float(row[3])
    if name in review_max and n_reviews> review_max[name]:
        review_max[name] = n_reviews
    elif name not in review_max:
            review_max[name] =  n_reviews
print(len(review_max))            

9659


In the previous code cell, we found that there are 1,181 cases where our app occurs more than once, so the length of our dictionary will be difference between our data set and 1,181.

In [10]:
print("Expected Length", len(android) - 1181)
print("Actual length", len(review_max))

Expected Length 9659
Actual length 9659


Now, let's use this dictionary to remove the duplicate data. For, the duplicate cases, we only keep the entries with maximum number of reviews.
* We start by initializing two empty lists, android_clean and already_added.
* We loop through the android data set, and for every iteration:
        * we assign the name of the app and the number of reviews.
        * We add the current row (app) to the android_clean list, and the app name (name) to the already_cleaned list if:
                 * The number of reviews of the current app matches the number of reviews of that app as described in the
                 reviews_max dictionary 
                 * The name of the app is not already in the already_added list list

In [11]:
android_clean = []
already_added = []
for row in android:
    name = row[0]
    review = float(row[3])
    if  (review_max[name] ==  review) and (name not in already_added) :  # checking the condition
        android_clean.append(row)
        already_added.append(name) 

Now, lets explore the new data set, and cross check whether the number of rows is 9659


In [12]:
explore_Data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows  9659
Number of columns  13


As expected we have 9659 rows.

# Removing Non-English apps


## Part One 

If you explore this data set, you'll notice the names of some of the apps suggest they are not directed toward an English-speaking audience. Below, we see a couple of examples from both data sets:

In [13]:
print(ios[814][2])
print(android_clean[4412][0])
print(android_clean[7940][0])

搜狐新闻—新闻热点资讯掌上阅读软件
中国語 AQリスニング
لعبة تقدر تربح DZ


We are not interested in keeping there kind of apps, so we will remove them. One way to remove each app whose name contains a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We built this function below, and we use the built-in ord() function to find out the corresponding encoding number of each character.

In [14]:
def is_English(text):
    for character in text:
        if ord(character) > 127:
            return False
    
    return True

print(is_English('Instagram'))
print(is_English('爱奇艺PPS -《欢乐颂2》电视剧热播')) 

True
False


This seems to work fine but some English apps use emojis, that fall outside ASCII range, because of this we will remove some useful apps.

## Part two 

To minimise the impact we can eliminate those apps who has more than three non-ASCII characters. 

In [15]:
def isEnglish(text):
    non_ascii = 0
    for character in text:
        if ord(character) >  127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else: 
        return True
    
print(isEnglish('Instagram'))
print(isEnglish('爱奇艺PPS -《欢乐颂2》电视剧热播'))     
print(isEnglish('Docs To Go™ Free Office Suite'))
print(isEnglish('Instachat 😜'))

True
False
True
True


Below, we use isEnglish() to filter out the non-english apps for both data sets

In [16]:
ios_English = []
android_English = []
for row in android_clean:
    name = row[0]
    if isEnglish(name):
        android_English.append(row)
        
for row in ios:
    name = row[2]
    if isEnglish(name):
        ios_English.append(row)
print(len(ios_English))
print(len(android_English))

6183
9614


We are left with 6184 ios apps and 9614 android apps

# Isolating the free apps

In [17]:
ios_final = []
android_final = []
for row in android_English:
    price = row[7]
    if price == "0":
        android_final.append(row)      
for row in ios_English:
    price = row[5]
    if price == "0":
        ios_final.append(row)
print(len(ios_final))
print(len(android_final))
        

3222
8864


We are left with 3222 ios apps, and 8864 android apps

# Most common apps by genre

Our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
* Build a minimal Android version of the app, and add it to Google Play.
* If the app has a good response from users, we then develop it further.
* If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

We'll build two functions we can use to analyze the frequency tables:
* One function to generate frequency tables that show percentages
* Another function that we can use to display the percentages in a descending order

In [18]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for row in dataset: 
        total+=1
        value = row[index]
        if value in table:
            table[value]+=1
        else:
            table[value] = 1        
    table_percentages = {}
    for row in table:
        percentage = (table[row]/total) * 100
        table_percentages[row] = percentage
    return table_percentages       

In [19]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_value_as_tuple = (table[key],key)
        table_display.append(key_value_as_tuple)
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1] ,": ", entry[0] )

In [20]:
display_table(ios_final, -5)

Games :  58.16263190564867
Entertainment :  7.883302296710118
Photo & Video :  4.9658597144630665
Education :  3.662321539416512
Social Networking :  3.2898820608317814
Shopping :  2.60707635009311
Utilities :  2.5139664804469275
Sports :  2.1415270018621975
Music :  2.0484171322160147
Health & Fitness :  2.0173805090006205
Productivity :  1.7380509000620732
Lifestyle :  1.5828677839851024
News :  1.3345747982619491
Travel :  1.2414649286157666
Finance :  1.1173184357541899
Weather :  0.8690254500310366
Food & Drink :  0.8069522036002483
Reference :  0.5586592178770949
Business :  0.5276225946617008
Book :  0.4345127250155183
Navigation :  0.186219739292365
Medical :  0.186219739292365
Catalogs :  0.12414649286157665


We can see among the english free apps, more than half of them are games. Entertainment apps are close to 8%, Photo and video apps are close to 5%, followed by Education and Social Networking. 

So in IOS apps (at least the part containing free English apps), we can conclude most of the apps which is dominating are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more only few.


Let's now continue by examining the genre and Category columns of the Google Play dataset.

In [22]:
display_table(android_final, 1)

FAMILY :  18.907942238267147
GAME :  9.724729241877256
TOOLS :  8.461191335740072
BUSINESS :  4.591606498194946
LIFESTYLE :  3.9034296028880866
PRODUCTIVITY :  3.892148014440433
FINANCE :  3.7003610108303246
MEDICAL :  3.531137184115524
SPORTS :  3.395758122743682
PERSONALIZATION :  3.3167870036101084
COMMUNICATION :  3.2378158844765346
HEALTH_AND_FITNESS :  3.0798736462093865
PHOTOGRAPHY :  2.944494584837545
NEWS_AND_MAGAZINES :  2.7978339350180503
SOCIAL :  2.6624548736462095
TRAVEL_AND_LOCAL :  2.33528880866426
SHOPPING :  2.2450361010830324
BOOKS_AND_REFERENCE :  2.1435018050541514
DATING :  1.861462093862816
VIDEO_PLAYERS :  1.7937725631768955
MAPS_AND_NAVIGATION :  1.3989169675090252
FOOD_AND_DRINK :  1.2409747292418771
EDUCATION :  1.1620036101083033
ENTERTAINMENT :  0.9589350180505415
LIBRARIES_AND_DEMO :  0.9363718411552346
AUTO_AND_VEHICLES :  0.9250902527075812
HOUSE_AND_HOME :  0.8235559566787004
WEATHER :  0.8009927797833934
EVENTS :  0.7107400722021661
PARENTING :  0.6543

The scenario looks different in Google Play:  there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:

In [23]:
display_table(android_final, -4)

Tools :  8.449909747292418
Entertainment :  6.069494584837545
Education :  5.347472924187725
Business :  4.591606498194946
Productivity :  3.892148014440433
Lifestyle :  3.892148014440433
Finance :  3.7003610108303246
Medical :  3.531137184115524
Sports :  3.463447653429603
Personalization :  3.3167870036101084
Communication :  3.2378158844765346
Action :  3.1024368231046933
Health & Fitness :  3.0798736462093865
Photography :  2.944494584837545
News & Magazines :  2.7978339350180503
Social :  2.6624548736462095
Travel & Local :  2.3240072202166067
Shopping :  2.2450361010830324
Books & Reference :  2.1435018050541514
Simulation :  2.0419675090252705
Dating :  1.861462093862816
Arcade :  1.8501805054151623
Video Players & Editors :  1.7712093862815883
Casual :  1.7599277978339352
Maps & Navigation :  1.3989169675090252
Food & Drink :  1.2409747292418771
Puzzle :  1.128158844765343
Racing :  0.9927797833935018
Role Playing :  0.9363718411552346
Libraries & Demo :  0.9363718411552346
Aut

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

In [24]:
genre_ios = freq_table(ios_final, -5)
print(genre_ios)

{'Productivity': 1.7380509000620732, 'Weather': 0.8690254500310366, 'Shopping': 2.60707635009311, 'Reference': 0.5586592178770949, 'Finance': 1.1173184357541899, 'Music': 2.0484171322160147, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Social Networking': 3.2898820608317814, 'Sports': 2.1415270018621975, 'Health & Fitness': 2.0173805090006205, 'Games': 58.16263190564867, 'Food & Drink': 0.8069522036002483, 'News': 1.3345747982619491, 'Book': 0.4345127250155183, 'Photo & Video': 4.9658597144630665, 'Entertainment': 7.883302296710118, 'Business': 0.5276225946617008, 'Lifestyle': 1.5828677839851024, 'Education': 3.662321539416512, 'Navigation': 0.186219739292365, 'Medical': 0.186219739292365, 'Catalogs': 0.12414649286157665}


In [30]:
for genre in genre_ios:
    total = 0
    length = 0 
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_rating = float(app[6])
            total += n_rating
            length +=1
    avg_n_Rating = total/length        
    print(genre, ":", avg_n_Rating)

Productivity : 21028.410714285714
Weather : 52279.892857142855
Shopping : 26919.690476190477
Reference : 74942.11111111111
Finance : 31467.944444444445
Music : 57326.530303030304
Utilities : 18684.456790123455
Travel : 28243.8
Social Networking : 71548.34905660378
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Games : 22788.6696905016
Food & Drink : 33333.92307692308
News : 21248.023255813954
Book : 39758.5
Photo & Video : 28441.54375
Entertainment : 14029.830708661417
Business : 7491.117647058823
Lifestyle : 16485.764705882353
Education : 7003.983050847458
Navigation : 86090.33333333333
Medical : 612.0
Catalogs : 4004.0


On average, navigation apps have the most number of user reviews, but this is heavily influenced by Google maps.

In [40]:
for app in ios_final:
    if app[-5]== "Navigation":     
        print(app[2],':', app[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


In [41]:
for app in ios_final:
    if app[-5]== "Social Networking":     
        print(app[2],':', app[6])

Facebook : 2974676
LinkedIn : 71856
Skype for iPhone : 373519
Tumblr : 334293
Match™ - #1 Dating App. : 60659
WhatsApp Messenger : 287589
TextNow - Unlimited Text + Calls : 164963
Grindr - Gay and same sex guys chat, meet and date : 23201
imo video calls and chat : 18841
Ameba : 269
Weibo : 7265
Badoo - Meet New People, Chat, Socialize. : 34428
Kik : 260965
Qzone : 1649
Fake-A-Location Free ™ : 354
Tango - Free Video Call, Voice and Chat : 75412
MeetMe - Chat and Meet New People : 97072
SimSimi : 23530
Viber Messenger – Text & Call : 164249
Find My Family, Friends & iPhone - Life360 Locator : 43877
Weibo HD : 16772
POF - Best Dating App for Conversations : 52642
GroupMe : 28260
Lobi : 36
WeChat : 34584
ooVoo – Free Video Call, Text and Voice : 177501
Pinterest : 1061624
知乎 : 397
Qzone HD : 458
Skype for iPad : 60163
LINE : 11437
QQ : 9109
LOVOO - Dating Chat : 1985
QQ HD : 5058
Messenger : 351466
eHarmony™ Dating App - Meet Singles : 11124
YouNow: Live Stream Video Chat : 12079
Cougar 

As we can see in Social Networking too, the average number is heavily influenced by few giants like Facebook, Whatspp messenger etc.
Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000

Reference apps have 74942 rating, let's examine that for a better picture

In [42]:
for app in ios_final:
    if app[-5]== "Reference":     
        print(app[2],':', app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


However, this seems to show some interesting facts, One thing which we could do is take a popular book and turn it into an app where we could add different features besides the raw version of book, this might include qaily quotes, some motivational quotes, and how about adding a dictionary too, so that user doesnt need to exit the apps.

Now, let's analyze Google dataset

## Most Popular apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [43]:
display_table(android_final, 5)

1,000,000+ :  15.726534296028879
100,000+ :  11.552346570397113
10,000,000+ :  10.548285198555957
10,000+ :  10.198555956678701
1,000+ :  8.393501805054152
100+ :  6.915613718411552
5,000,000+ :  6.825361010830325
500,000+ :  5.561823104693141
50,000+ :  4.7721119133574
5,000+ :  4.512635379061372
10+ :  3.5424187725631766
500+ :  3.2490974729241873
50,000,000+ :  2.3014440433213
100,000,000+ :  2.1322202166064983
50+ :  1.917870036101083
5+ :  0.78971119133574
1+ :  0.5076714801444043
500,000,000+ :  0.2707581227436823
1,000,000,000+ :  0.22563176895306858
0+ :  0.04512635379061372
0 :  0.01128158844765343


In [45]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [46]:
for app in android_final:
    if app[1] == "COMMUNICATION" and (app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+'):
        print(app[0], ":", app[5])
    

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Here also we can see "Communication" is being dominated by Whatsapp messenger, Google duo and all, if we remove these gigantic apps, the average would be reduced alot.
We see the same pattern for VIDEO_PLAYERS which is being dominated by Youtube and Google Play movies.

The main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.
The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [47]:
for app in android_final:
    if app[1] == "BOOKS_AND_REFERENCE":
        print(app[0], ":", app[5])
    

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. 

Let's examine more as we need have already shown some interest for this genre in AppleStore apps.

In [48]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads)

In [52]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '500,000+'
                                            or app[5] == '100,000+'    
                                            or app[5] == '50,000+' ):
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Litnet - E-books : 100,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
English translation from Bengali : 100,000+
Pdf Book Download - Read Pdf Book : 100,000+
Free Book Reader : 100,000+
eBoox new: Reader for fb2 epub zip books : 50,000+
Only 30 days in English, the guideline is guaranteed : 500,000+
SH-02J Owner's Manual (Android 8.0) : 50,000+
English-Myanmar Dictionary : 1,000,0

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

# Conclusion

In this project we analyzed data about the App store and Play store mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book)  and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, dictionaries etc.